### PR TITLE
[WIP][SYCL] Replaces some of the CL_* enums with PI_* enums.

### PR DESCRIPTION
**Pull request by rbegam**
Signed-off-by: Rehana Begam <rehana.begam@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/pi.h [hunk 1]
```diff
@@ -365,25 +424,43 @@ const pi_queue_properties PI_QUEUE_ON_DEVICE_DEFAULT =
...
+  PI_DEVICE_PARTITION_EQUALLY = CL_DEVICE_PARTITION_EQUALLY;
+static const pi_device_partition_property
+  PI_DEVICE_PARTITION_BY_AFFINITY_DOMAIN = CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN;
+
+// For compatibility with OpenCL define this not as enum.
+typedef pi_bitfield pi_device_affinity_domain;
+static const pi_device_affinity_domain
+  PI_DEVICE_AFFINITY_DOMAIN_NUMA = CL_DEVICE_AFFINITY_DOMAIN_NUMA;
+static const pi_device_affinity_domain
+  PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE = CL_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE;
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.h]:
```suggestion
using pi_device_partition_property intptr_t;
static constexpr pi_device_partition_property
  PI_DEVICE_PARTITION_EQUALLY = CL_DEVICE_PARTITION_EQUALLY;
static constexpr pi_device_partition_property
  PI_DEVICE_PARTITION_BY_AFFINITY_DOMAIN = CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN;

// For compatibility with OpenCL define this not as enum.
using pi_device_affinity_domain pi_bitfield;
static constexpr pi_device_affinity_domain
  PI_DEVICE_AFFINITY_DOMAIN_NUMA = CL_DEVICE_AFFINITY_DOMAIN_NUMA;
static constexpr pi_device_affinity_domain
  PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE = CL_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE;
```

File: sycl/include/CL/sycl/detail/pi.h [hunk 2]
```diff
@@ -332,14 +391,14 @@ typedef enum {
...
-  PI_EVENT_INFO_CONTEXT                  = CL_EVENT_CONTEXT
-} _pi_event_info;
+typedef pi_bitfield pi_device_exec_capabilities;
+const pi_device_exec_capabilities PI_DEVICE_EXEC_CAPABILITIES_KERNEL        = CL_EXEC_KERNEL;
+const pi_device_exec_capabilities PI_DEVICE_EXEC_CAPABILITIES_NATIVE_KERNEL = CL_EXEC_NATIVE_KERNEL;
 
+typedef pi_bitfield pi_sampler_properties;
+const pi_sampler_properties PI_SAMPLER_PROPERTIES_NORMALIZED_COORDS = CL_SAMPLER_NORMALIZED_COORDS;
+const pi_sampler_properties PI_SAMPLER_PROPERTIES_ADDRESSING_MODE   = CL_SAMPLER_ADDRESSING_MODE;
+const pi_sampler_properties PI_SAMPLER_PROPERTIES_FILTER_MODE       = CL_SAMPLER_FILTER_MODE;
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.h]:
```suggestion
using pi_device_exec_capabilities pi_bitfield;
constexpr pi_device_exec_capabilities PI_DEVICE_EXEC_CAPABILITIES_KERNEL        = CL_EXEC_KERNEL;
constexpr pi_device_exec_capabilities PI_DEVICE_EXEC_CAPABILITIES_NATIVE_KERNEL = CL_EXEC_NATIVE_KERNEL;

using pi_sampler_properties pi_bitfield;
constexpr pi_sampler_properties PI_SAMPLER_PROPERTIES_NORMALIZED_COORDS = CL_SAMPLER_NORMALIZED_COORDS;
constexpr pi_sampler_properties PI_SAMPLER_PROPERTIES_ADDRESSING_MODE   = CL_SAMPLER_ADDRESSING_MODE;
constexpr pi_sampler_properties PI_SAMPLER_PROPERTIES_FILTER_MODE       = CL_SAMPLER_FILTER_MODE;
```

**Comment by romanovvlad** [comment]
@rbegam Please, remove commits not related to your patch form PR.

**Comment by rbegam** [comment]
@romanovvlad @bader @garimagu 
I've created a fresh PR for this change here: https://github.com/intel/llvm/pull/1239
It contains modifications requested so far. Please review that one. 

I'll close this one soon.

**Pull request closed without merging**